dockhub.py

```diff
@@ -1,226 +1,182 @@
 #!/usr/bin/env python3
 
+from functools import wraps
 import json
 from os import getenv
-from sys import exit
+import sys
 
 import click
 import requests
 
 
-def die(msg=None, exception=None):
-    if msg is not None:
-        print(f'{msg}')
-    if exception is not None:
-        print(f'{exception}')
-    exit(1)
+CONTENT_HEADER = {'Content-Type': 'application/json', 'charset': 'utf-8'}
+DOCKERHUB_URL = 'https://hub.docker.com/v2'
 
 
+def handle_http_errors(fun):
```

Comment:
In Python, a function decorator is a function that takes another function as an argument. This lets you wrap functions with functionality. I think it's helpful here to centralize handling for the various http errors.

There are other ways to reduce the redundancy in error handling that you were concerned about. We could put the handling in `main`. We could create a function that calls `requests.get` with a given url and parameters and handles errors including the non-200 status code and returns the response if it's good. That might be a better way now that I'm looking at this.

Maybe something like this:

```
def request_get(url, params=None, headers=None):
    try:
        resp = request_get(url, params=params, headers=headers)
        resp.raise_for_status()
    except requests.ConnectionError as exc:
        ...
```

Then you'd use it like this:

```
     group = requests_get(docker_api_url, headers=auth_header)
```

I kind of like that.

What feels better to you?